src/main/thrift/parquet.thrift

```diff
@@ -595,6 +607,43 @@ union ColumnOrder {
   1: TypeDefinedOrder TYPE_ORDER;
 }
 
+struct PageLocation {
+/** Offset of the page in the file **/
+  1: required i64 offset
+
+/** Size of the page, including header. The same as PageHeader.compressed_page_size **/
+  2: required i32 compressed_page_size
+
+/** Index within the RowGroup of the first row of the page **/
+  3: required i64 first_row_index
+}
+
+struct OffsetIndex {
+/** PageLocations, ordered by increasing PageLocation.offset **/
+  1: required list<PageLocation> page_locations
+}
+
+/**
+ * Description for ColumnIndex.
+ * Each <array-field>[i] refers to the page at OffsetIndex.page_locations[i]
+ */
+struct ColumnIndex {
+/** A list of bools to determine the validity of the corresponding min and max values **/
+  1: required list<bool> valid_values
```

Comment:
I think we should rename this to be more clear. From my notes, this is true if there was a non-null value so the min and max in the next two lists are also non-null.

How about naming this `null_pages`? This should also state that the corresponding `min_values` and `max_values` entries for null pages are `byte[0]`.

Reply:
+1 for null_pages.